app-react/src/jsMain/kotlin/com/ustadmobile/view/message/messagelist/MessageListScreen.kt

```diff
@@ -0,0 +1,181 @@
+            width = 100.pct
+            contain = Contain.strict
+            overflowY = Overflow.scroll
+        }
+
+        content = virtualListContent {
+
+            infiniteQueryPagingItems(
+                items = infiniteQueryResult,
+                key = { it.messageUid.toString() }
+            ) { messageItem ->
+                ChatItem.create {
+                    message = messageItem
+                    activeUserUid = props.uiState.activePersonUid
+                }
+            }
+        }
+
+        Container {
+            VirtualListOutlet()
+        }
+    }
+
+    Container {
+        ref = newMessageBoxRef
+
+        UstadSendTextField {
+            value = props.uiState.newMessageText
+            fullWidth = true
+            onTextChange = props.onChangeNewMessageText
+            onClickSend = props.onClickSend
+        }
+    }
+}
+
+val MessageListScreenPreview = FC<Props> {
+
+    val messages = { ListPagingSource(
+        listOf(Message().apply {
+            messageText = "Sallam, WHere are you from?"
+            messageSenderPersonUid = 1
+        },
+            Message().apply {
+                messageText = "Sallam, WHere are you from?"
+                messageSenderPersonUid = 2
+            },
+        )
+    ) }
+
+    val MessageListUiState by useState {
+        MessageListUiState(
+            messages = messages,
+            activePersonUid = 2
+        )
+    }
+
+    MessageListScreenComponent2 {
+        uiState = MessageListUiState
+    }
+}
+
+val MessageListScreen = FC<Props> {
+    val viewModel = useUstadViewModel { di, savedStateHandle ->
+        MessageListViewModel(di, savedStateHandle)
+    }
+    val uiStateVal by viewModel.uiState.collectAsState(MessageListUiState())
+
+    MessageListScreenComponent2 {
+        uiState = uiStateVal
+        onChangeNewMessageText = viewModel::onChangeNewMessageText
+        onClickSend = viewModel::onClickSend
+    }
+}
+
+external interface ChatItemProps: Props {
+
+    var message: Message?
+    var activeUserUid: Long
+
+}
+
+val ChatItem = FC<ChatItemProps> { props ->
+
+    val theme by useRequiredContext(ThemeContext)
+    val isFromMe = props.message?.messageSenderPersonUid == props.activeUserUid
+
+    Box {
+        sx {
+            display = Display.flex
+            flex = Flex.minContent
+            justifyContent = if (isFromMe) JustifyContent.end else JustifyContent.start
+        }
+
+        Typography {
+            sx {
+                backgroundColor = theme.palette.primary.light
+                color = theme.palette.primary.contrastText
+                padding = 10.px
+                margin = 5.px
+                borderTopLeftRadius = 48.px
+                borderTopRightRadius = 48.px
+                borderBottomLeftRadius = if (isFromMe) 48.px else 0.px
+                borderBottomRightRadius = if (isFromMe) 0.px else 48.px
+            }
+            + UstadLinkify.create {
```

Comment:
In react we have two ways of adding components: passing a ReactNode object and invoking the component function. For example on label we have something like:

```
TextField {
    label = ReactNode("label text")
}
```

If we don't want to use plain text, then a React Node object can be created using .create e.g. 
```
TextField {
   label = Typography.create {
       ...
   }
}
```

Inside another node ("ChildrenBuilder" scope), then :

```
+ ComponentType.create { 
   propName = "value"
} 
```
Is the same as 
```
ComponentType {
    propName = "value"
} 
```

So when adding a component to another component that supports children, we would normally not use + ComponentType.create { }, use use ComponentType { .. }